cudamapper/src/main.cu

```diff
@@ -226,13 +247,17 @@ int main(int argc, char* argv[])
     // Round 2
     // Query: [1000-1999] - Use cache entry (from previous use when now query was a target)
     // Etc..
-    auto evict_index = [&index_cache](const claragenomics::cudamapper::read_id_t query_start_index,
-                                      const claragenomics::cudamapper::read_id_t query_end_index,
-                                      const int device_id) {
+    auto evict_index = [&device_index_cache, &host_index_cache](const claragenomics::cudamapper::read_id_t query_start_index,
+                                                                const claragenomics::cudamapper::read_id_t query_end_index,
+                                                                const int device_id,
+                                                                const int num_devices) {
         std::pair<uint64_t, uint64_t> key;
         key.first  = query_start_index;
         key.second = query_end_index;
-        index_cache[device_id].erase(key);
+        device_index_cache[device_id].erase(key);
+        // ToDo: freeing host memory is limited to 1 GPU, need to change to work with multiple GPUs
```

Comment:
Same as above (if not too much work please enable multi-GPU)

Reply:
the RAM eventually will be freed at the end, but this line frees up memory gradually as computation progresses forward. Since freeing-up memory depends on access pattern by different GPUs (preferrably we don't want to earse a key where may be accessed by another GPU sometime later), I kept it as it is and added this comment:
`// host memory can be freed by removing (key) when working with 1 GPU`
`// in multiple GPUs we keep (key), as it may be accessed by other GPUs depending on access pattern`